test/Dialect/LLHD/Transforms/desequentialization.mlir

```diff
@@ -0,0 +1,111 @@
+  // CHECK: llhd.drv [[SIG]], [[V2]] after
+  %time = llhd.constant_time <0ns, 1d, 0e>
+  %false = hw.constant false
+  %true = hw.constant true
+  llhd.process {
+    cf.br ^bb1
+  ^bb1:
+    %p1 = llhd.prb %clk : !hw.inout<i1>
+    llhd.wait (%p1 : i1), ^bb2
+  ^bb2:
+    %p2 = llhd.prb %clk : !hw.inout<i1>
+    %old = comb.xor %p1, %true : i1
+    %posedge = comb.and %old, %p2 : i1
+    llhd.drv %sig, %false after %time if %posedge : !hw.inout<i1>
+    cf.br ^bb1
+  }
+}
+
+// CHECK-LABEL: @negedgeClk
+// CHECK-SAME: (inout [[CLK:%.+]] : i1, inout [[SIG:%.+]] : i1)
+hw.module @negedgeClk(inout %clk : i1, inout %sig : i1) {
+  // CHECK: [[V0:%.+]] = llhd.prb [[CLK]]
+  // CHECK: [[V1:%.+]] = seq.to_clock [[V0]]
+  // CHECK: [[V2:%.+]] = seq.clock_inv [[V1]]
+  // CHECK: [[V3:%.+]] = seq.compreg %false{{.*}}, [[V2]]
+  // CHECK: llhd.drv [[SIG]], [[V3]] after
+  %time = llhd.constant_time <0ns, 1d, 0e>
+  %false = hw.constant false
+  %true = hw.constant true
+  llhd.process {
+    cf.br ^bb1
+  ^bb1:
+    %p1 = llhd.prb %clk : !hw.inout<i1>
+    llhd.wait (%p1 : i1), ^bb2
+  ^bb2:
+    %p2 = llhd.prb %clk : !hw.inout<i1>
+    %old = comb.xor %p2, %true : i1
+    %posedge = comb.and %old, %p1 : i1
+    llhd.drv %sig, %false after %time if %posedge : !hw.inout<i1>
+    cf.br ^bb1
+  }
+}
+
+// CHECK-LABEL: @enable
+// CHECK-SAME: (in [[C:%.+]] : i1, inout [[CLK:%.+]] : i1, inout [[SIG:%.+]] : i1)
+hw.module @enable(in %c : i1, inout %clk : i1, inout %sig : i1) {
+  // CHECK: [[V0:%.+]] = llhd.prb [[CLK]]
+  // CHECK: [[V1:%.+]] = seq.to_clock [[V0]]
+  // CHECK: [[V2:%.+]] = seq.clock_gate [[V1]], [[C]]
+  // CHECK: [[V3:%.+]] = seq.compreg %false{{.*}}, [[V2]]
+  // CHECK: llhd.drv [[SIG]], [[V3]] after
+  %time = llhd.constant_time <0ns, 1d, 0e>
+  %false = hw.constant false
+  %true = hw.constant true
+  llhd.process {
+    cf.br ^bb1
+  ^bb1:
+    %p1 = llhd.prb %clk : !hw.inout<i1>
+    llhd.wait (%p1 : i1), ^bb2
+  ^bb2:
+    %p2 = llhd.prb %clk : !hw.inout<i1>
+    %old = comb.xor %p1, %true : i1
+    %posedge = comb.and %old, %p2 : i1
+    %cond = comb.and %posedge, %c : i1
+    llhd.drv %sig, %false after %time if %cond : !hw.inout<i1>
+    cf.br ^bb1
+  }
+}
+
+// CHECK-LABEL: @asyncReset
+// CHECK-SAME: (inout [[RST:%.+]] : i1, inout [[CLK:%.+]] : i1, inout [[SIG:%.+]] : i1)
+hw.module @asyncReset(inout %rst : i1, inout %clk : i1, inout %sig : i1) {
+  // CHECK: [[V0:%.+]] = llhd.prb [[CLK]]
+  // CHECK: [[V1:%.+]] = llhd.prb [[RST]]
+  // CHECK: [[V2:%.+]] = seq.to_clock [[V0]]
+  // CHECK: [[V3:%.+]] = comb.xor [[V1]], %true{{.*}}
+  // CHECK: [[V4:%.+]] = seq.compreg %false{{.*}}, [[V2]] reset [[V3]], %false{{.*}}
+  // CHECK: llhd.drv [[SIG]], [[V4]] after
+  %time = llhd.constant_time <0ns, 1d, 0e>
+  %false = hw.constant false
+  %true = hw.constant true
+  llhd.process {
+    cf.br ^bb1
+  ^bb1:
+    %p1 = llhd.prb %clk : !hw.inout<i1>
+    %r1 = llhd.prb %rst : !hw.inout<i1>
+    llhd.wait (%p1 : i1), ^bb2
```

Comment:
I think this wait needs to also observe the reset in order to make the reset async. Otherwise this would be more like a sync reset.